### PR TITLE
HC-15: Doctor's filtration panel component

### PR DESCRIPTION
**Pull request by irishdra**

### CODE REVIEW

File: frontend/src/app.tsx [hunk 1]
```diff
@@ -4,11 +4,13 @@ import { AppRoute } from 'common/enums';
 import Sign from 'components/sign/sign'
 import NotFound from 'components/not-found/not-found';
 import DoctorsSearch from 'components/doctors-search/doctors-search';
+import DoctorsFiltration from 'components/doctors-filtration-panel/doctors-filtration-panel';
 
 const App: React.FC = () => (
   <Switch>
     <Route path={[AppRoute.SIGN_IN, AppRoute.SIGN_UP]} component={Sign}/>
     <Route path={AppRoute.DOCTORS} component={DoctorsSearch}/>
+    <Route path="/" component={DoctorsFiltration}/>
```
Comment (ghost) [inline frontend/src/app.tsx]:
```suggestion
    <Route path={AppRoute.ROOT} component={DoctorsFiltration}/>
```

File: frontend/src/components/common/details/details.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  icon: string;
+  title: string;
+  children: React.ReactNode;
+}
+
+const Details: React.FC<Props> = ({ icon, title, children }) => {
+  return (
+    <details className={clsx(styles.commonFilter)}>
+      <summary className={clsx(styles.filterInfo)}>
+        <img src={icon} loading="lazy" alt="Filter icon."/>
```
Comment (ghost) [inline frontend/src/components/common/details/details.tsx]: Make this icon as background img

File: frontend/src/components/common/details/details.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import * as React from 'react';
+import clsx from 'clsx';
+import styles from './styles.module.scss';
+
+interface Props {
+  icon: string;
+  title: string;
+  children: React.ReactNode;
```
Comment (ghost) [inline frontend/src/components/common/details/details.tsx]:
```suggestion
```
React.FC will do it for you

File: frontend/src/components/doctors-filtration-panel/doctors-filtration-panel.tsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+  [ClinicType.STATE]: false
+}
+
+const DoctorsFiltration: React.FC = () => {
+  const [checkboxesValues, setCheckboxesValues] = React.useState(CHECKBOXES_DEFAULT_VALUES);
+
+  const { handleSubmit, control, errors } = useForm<IDoctorFiltrationPayload>({
+    defaultValues: DEFAULT_VALUES,
+    mode: "onChange"
+  });
```
Comment (ghost) [inline frontend/src/components/doctors-filtration-panel/doctors-filtration-panel.tsx]:
```suggestion
  const [checkboxesValues, setCheckboxesValues] = React.useState(CHECKBOXES_DEFAULT_VALUES);

  const { handleSubmit, control, errors } = useForm<IDoctorFiltrationPayload>({
    defaultValues: DEFAULT_VALUES,
    mode: "onChange"
  });
```
we need to use only `useForm` without any other input handlers

File: frontend/src/components/doctors-filtration-panel/doctors-filtration-panel.tsx [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+  const onSubmit = (data: IDoctorFiltrationPayload) => {
+    console.log(data);
+  };
+
+  const onCheckboxChange = (name: keyof IDoctorCheckbox) => {
+    setCheckboxesValues({
+      ...checkboxesValues,
+      [name]: !checkboxesValues[name]
+    })
+  }
```
Comment (ghost) [inline frontend/src/components/doctors-filtration-panel/doctors-filtration-panel.tsx]:
```suggestion
  const onSubmit = (data: IDoctorFiltrationPayload) => {
    console.log(data);
  };

  const onCheckboxChange = (name: keyof IDoctorCheckbox) => {
    setCheckboxesValues({
      ...checkboxesValues,
      [name]: !checkboxesValues[name]
    })
  }
```
use handle<Element><Event> for local component cb's

File: frontend/src/components/doctors-filtration-panel/doctors-filtration-panel.tsx [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+            errors={errors}
+          />
+        </Details>
+        <Details
+          icon={specialty}
+          title={DoctorFiltration.SPECIALTY}
+        >
+          <div className={styles.filterCheckbox}>
+            <Checkbox
+              onChange={() => onCheckboxChange(DoctorType.PEDIATRICIAN)}
```
Comment (ghost) [inline frontend/src/components/doctors-filtration-panel/doctors-filtration-panel.tsx]: Don't use anonymous functions inside `jsx`

File: shared/src/common/enums/doctor/doctor-assessment.enum.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
+enum DoctorAssessment {
```
Comment (ghost) [inline shared/src/common/enums/doctor/doctor-assessment.enum.ts]: We don't use so remove it

File: shared/src/common/enums/filtration/doctor-filtration.enum.ts [hunk 8]
```diff
@@ -0,0 +1,9 @@
+enum DoctorFiltration {
```
Comment (ZubkoWhatislove) [inline shared/src/common/enums/filtration/doctor-filtration.enum.ts:1]: Remove values that we don't use

File: shared/src/common/enums/payment/payment-type.enum.ts [hunk 9]
```diff
@@ -0,0 +1,6 @@
+enum PaymentType {
```
Comment (ghost) [inline shared/src/common/enums/payment/payment-type.enum.ts]: We don't use it, so remove it

File: shared/src/common/interfaces/filtration/doctor-checkbox.interface.ts [hunk 10]
```diff
@@ -0,0 +1,13 @@
+import { DoctorType, ClinicType } from '../../enums';
+
+interface IDoctorCheckbox {
```
Comment (ghost) [inline shared/src/common/interfaces/filtration/doctor-checkbox.interface.ts]: We don't use it, so remove it

File: shared/src/common/interfaces/filtration/doctor-filtration-payload.interface.ts [hunk 11]
```diff
@@ -0,0 +1,10 @@
+import { DoctorFiltration, DoctorType, ClinicType } from '../../enums';
+
+interface IDoctorFiltrationPayload {
+  [DoctorFiltration.SEARCH]: string;
+  [DoctorFiltration.CITY]: string;
+  [DoctorFiltration.SPECIALTY]: DoctorType;
```
Comment (ghost) [inline shared/src/common/interfaces/filtration/doctor-filtration-payload.interface.ts]:
```suggestion
  [DoctorFiltration.SPECIALTY]: DoctorType[];
```

File: shared/src/common/interfaces/filtration/doctor-filtration-payload.interface.ts [hunk 12]
```diff
@@ -0,0 +1,10 @@
+import { DoctorFiltration, DoctorType, ClinicType } from '../../enums';
```
Comment (ghost) [inline shared/src/common/interfaces/filtration/doctor-filtration-payload.interface.ts]:
```suggestion
import { DoctorFiltration, DoctorType, ClinicType } from '~/enums';
```

File: shared/src/common/interfaces/filtration/doctor-checkbox.interface.ts [hunk 13]
```diff
@@ -0,0 +1,13 @@
+import { DoctorType, ClinicType } from '../../enums';
```
Comment (ghost) [inline shared/src/common/interfaces/filtration/doctor-checkbox.interface.ts]:
```suggestion
import { DoctorType, ClinicType } from '~/enums';
```

**Review by ghost — CHANGES_REQUESTED** [review]
On any form change the filter should return values something like:

```typescript
{
  doctorName: 'Joi', // 1
  city: '', // 2
  cpecialties: ['padiatrian'], // 3 (array only with selected values),
  typeOfClinic: 'state', // 4
}
```
image

File: frontend/src/components/common/details/details.tsx [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+
+const Details: React.FC<Props> = ({ icon, title, children }) => {
+  return (
+    <details className={styles.commonFilter}>
+      <summary className={styles.filterInfo}>
+        <div
+          className={styles.icon}
+          style={{
+            backgroundImage: `url(${icon})`
+          }}
```
Comment (ghost) [inline frontend/src/components/common/details/details.tsx]: use class for background image

File: frontend/src/components/doctors-filtration-panel/doctors-filtration-panel.tsx [hunk 15]
```diff
@@ -0,0 +1,126 @@
...
+
+  return (
+    <form onSubmit={handleSubmit(handleSubmitForm)}>
+    <div className={styles.panel}>
+      <div className={styles.filters}>
+        <div className={styles.filterHeader}>Filter by</div>
+        <div className={styles.commonFilter}>
+          <TextInput
+            name={DoctorFiltration.DOCTOR_NAME}
+            label=""
```
Comment (ghost) [inline frontend/src/components/doctors-filtration-panel/doctors-filtration-panel.tsx]: Always add labels

File: frontend/src/components/doctors-filtration-panel/doctors-filtration-panel.tsx [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+            errors={errors}
+            />
+        </div>
+        <Details
+          icon={location}
+          title="location"
+        >
+          <TextInput
+            name={DoctorFiltration.CITY}
+            label=""
```
Comment (ghost) [inline frontend/src/components/doctors-filtration-panel/doctors-filtration-panel.tsx]: Always add labels

File: frontend/src/components/common/details/details.tsx [hunk 17]
```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+import styles from './styles.module.scss';
+
+interface Props {
+  icon: string;
```
Comment (ghost) [inline frontend/src/components/common/details/details.tsx]: Create an enum `Icon` add all icons that we use in our application, and by the class, add icon properly

File: shared/src/common/enums/filtration/doctor-filtration.enum.ts [hunk 18]
```diff
@@ -0,0 +1,8 @@
...
+  DOCTOR_NAME = 'search by doctor name',
+  CITY = 'city',
+  SPECIALTY = 'specialty',
+  TYPE_OF_CLINIC = 'type of clinic'
+}
```
Comment (ghost) [inline shared/src/common/enums/filtration/doctor-filtration.enum.ts]:
```suggestion
enum DoctorFiltrationKey {
  DOCTOR_NAME = 'doctorName',
  CITY = 'city',
  SPECIALTY = 'specialty',
  TYPE_OF_CLINIC = 'typeOfClinic'
}
```

File: frontend/src/components/doctors-filtration-panel/doctors-filtration-panel.tsx [hunk 19]
```diff
@@ -0,0 +1,126 @@
...
+              label={ClinicType.PRIVATE}
+              control={control}
+            />
+          </div>
+            <div className={styles.filterCheckbox}>
+            <Checkbox
+              name={ClinicType.STATE}
+              label={ClinicType.STATE}
+              control={control}
+            />
```
Comment (ghost) [inline frontend/src/components/doctors-filtration-panel/doctors-filtration-panel.tsx]:
```suggestion
     const clinicTypes = Object.keys(ClinicType)


     {clinicTypes.map((clinicType) => (
            <Checkbox
              name={clinicType}
              label={clinicType}
              control={control}
            />
     ))}
```

Also do the same for checkboxes

File: frontend/src/components/doctors-filtration-panel/doctors-filtration-panel.tsx [hunk 20]
```diff
@@ -0,0 +1,126 @@
...
+import { DoctorType, ClinicType, InputType, InputColor, DoctorFiltration } from 'common/enums';
+import { IDoctorFiltrationPayload } from 'common/interfaces';
+import Details from 'components/common/details/details';
+
+const DEFAULT_VALUES: IDoctorFiltrationPayload = {
+  [DoctorFiltration.DOCTOR_NAME]: '',
+  [DoctorFiltration.CITY]: '',
+  [DoctorFiltration.SPECIALTY]: [],
+  [DoctorFiltration.TYPE_OF_CLINIC]: ClinicType.STATE
+}
```
Comment (ghost) [inline frontend/src/components/doctors-filtration-panel/doctors-filtration-panel.tsx]: Move it to the `frontend/src/components/doctors-filtration-panel/common/constants`

**Review by ghost — CHANGES_REQUESTED** [review]
👍

File: frontend/src/app.tsx [hunk 21]
```diff
@@ -6,6 +6,7 @@ import Sign from 'components/sign/sign'
 import NotFound from 'components/not-found/not-found';
 import Clinics from 'components/clinics/clinics';
 import DoctorsSearch from 'components/doctors-search/doctors-search';
+import DoctorsFiltration from 'components/doctors-filtration-panel/doctors-filtration-panel';
```
Comment (ghost) [inline frontend/src/app.tsx]: It should be part of doctors page
See design

File: frontend/src/common/enums/icons/icons.enum.ts [hunk 22]
```diff
@@ -0,0 +1,19 @@
+enum Icons {
```
Comment (ghost) [inline frontend/src/common/enums/icons/icons.enum.ts]: Enum name should be singular

File: frontend/src/components/common/details/details.tsx [hunk 23]
```diff
@@ -0,0 +1,24 @@
+import * as React from 'react';
+import clsx from 'clsx';
+import styles from './styles.module.scss';
+
+interface Props {
+  icon: string;
```
Comment (ghost) [inline frontend/src/components/common/details/details.tsx]:
```suggestion
  icon: Icon;
```

File: frontend/src/components/doctors-filtration-panel/doctors-filtration-panel.tsx [hunk 24]
```diff
@@ -0,0 +1,90 @@
...
+const clinicTypes = Object.keys(ClinicType);
+
+const DoctorsFiltration: React.FC = () => {
+  const { handleSubmit, control, errors } = useForm<IDoctorFiltrationPayload>({
+    defaultValues: DEFAULT_VALUES,
+    mode: "onChange",
+  });
+
+  const handleSubmitForm = (formData: IDoctorFiltrationPayload) => {
+    /* eslint-disable-next-line no-console */
```
Comment (ghost) [inline frontend/src/components/doctors-filtration-panel/doctors-filtration-panel.tsx]: Check **B13** criteria

File: frontend/src/components/doctors-filtration-panel/doctors-filtration-panel.tsx [hunk 25]
```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+import { useForm } from 'react-hook-form';
+import styles from './doctors-filtration.module.scss';
+import { TextInput, Checkbox } from 'components/common';
+import { DoctorType, ClinicType, InputType, InputColor, DoctorFiltration, Icons } from 'common/enums';
+import { IDoctorFiltrationPayload } from 'common/interfaces';
+import { DEFAULT_VALUES } from './common/constants/payload-default-values'
+import Details from 'components/common/details/details';
```
Comment (ghost) [inline frontend/src/components/doctors-filtration-panel/doctors-filtration-panel.tsx]:
```suggestion
import { TextInput, Checkbox, Details } from 'components/common';
import { DoctorType, ClinicType, InputType, InputColor, DoctorFiltration, Icons } from 'common/enums';
import { IDoctorFiltrationPayload } from 'common/interfaces';
import { DEFAULT_VALUES } from './common/constants/payload-default-values'
```

File: frontend/src/components/doctors-filtration-panel/doctors-filtration-panel.tsx [hunk 26]
```diff
@@ -0,0 +1,90 @@
...
+  return (
+    <form onSubmit={handleSubmit(handleSubmitForm)}>
+      <div className={styles.panel}>
+        <div className={styles.filters}>
+          <div className={styles.filterHeader}>Filter by</div>
+          <div className={styles.commonFilter}>
+            <TextInput
+              name={DoctorFiltration.DOCTOR_NAME}
+              label="Search by doctor's name"
+              hasHiddenLabel={false}
```
Comment (ghost) [inline frontend/src/components/doctors-filtration-panel/doctors-filtration-panel.tsx]: We non't have label in the design
```suggestion
              label="Search by doctor's name"
              hasHiddenLabel
```

File: frontend/src/components/doctors-filtration-panel/doctors-filtration-panel.tsx [hunk 27]
```diff
@@ -0,0 +1,90 @@
...
+            />
+          </div>
+          <Details
+            icon={Icons.LOCATION}
+            title="Location"
+          >
+            <TextInput
+              name={DoctorFiltration.CITY}
+              label={DoctorFiltration.CITY}
+              hasHiddenLabel={false}
```
Comment (ghost) [inline frontend/src/components/doctors-filtration-panel/doctors-filtration-panel.tsx]:
```suggestion
              hasHiddenLabel
```

**Review by ghost — CHANGES_REQUESTED** [review]
👍

**Review by ghost — DISMISSED** [review]
LGTM 👍

File: frontend/src/components/clinics/clinics.tsx [hunk 28]
```diff
@@ -1,5 +1,6 @@
 import * as React from 'react';
 import Clinic from './components/clinic/clinic';
+import DoctorsFiltration from '../doctors-filtration-panel/doctors-filtration-panel';
```
Comment (ghost) [inline frontend/src/components/clinics/clinics.tsx]:
```suggestion
import { Filtration } from './components';
```
Also rename component/file and etc

**Review by ghost — DISMISSED** [review]
👍

File: frontend/src/components/filtration/filtration.tsx [hunk 29]
```diff
@@ -0,0 +1,85 @@
...
+                />
+              </div>
+            ))}
+          </Details>
+        </div>
+      </div>
+    </form>
+  )}
+
+export { Filtration };
```
Comment (ZubkoWhatislove) [inline frontend/src/components/filtration/filtration.tsx]:
```suggestion
export default Filtration;
```

**Review by ghost — APPROVED** [review]
LGTM 👍

**Pull request merged**